sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -161,17 +161,16 @@ internal class AmqpSender : TransportSender
             CreateMessageBatchOptions options,
             CancellationToken cancellationToken)
         {
-            TransportMessageBatch messageBatch = null;
-            Task createBatchTask = _retryPolicy.RunOperation(async (timeout) =>
-            {
-                messageBatch = await CreateMessageBatchInternalAsync(
-                    options,
-                    timeout).ConfigureAwait(false);
-            },
-            _connectionScope,
-            cancellationToken);
-            await createBatchTask.ConfigureAwait(false);
-            return messageBatch;
+            return await _retryPolicy.RunOperation(static async (value, timeout, _) =>
```

Comment:
This is a bit of an odd one. We use `ValueTask` on the internal API and on the external API but the retry policy doesn't support `ValueTask`. Would it be worth to introduce a redundant path in the retry policy that is value task based?

Reply:
I wonder if we should switch the main path to be ValueTask-based so we can avoid allocating extra Task<T>.

Reply:
If we do so we can no longer avoid the async/await keywords. I'm fine with that. I push a version. Let me know if that suits better. Otherwise I can always revert

Reply:
Yes, I think it's better. We can switch AmqpSender/AmqpReceiver to ValueTasks and remove awaits later. This leaves us in a good place for future optimizations.

Reply:
Good for me